### PR TITLE
Library for periodic badges

### PR DESCRIPTION
**Pull request by Two9A**
### Summary
Probably makes it easier to add new special periods. There might be a way to merge in the room identifiers, but you may want those in room_identifiers separately anyways.

### Test Plan
Hah.

### CODE REVIEW

File: modules/token_collecting.js [hunk 1]
```diff
@@ -256,20 +257,10 @@ class TokenCollecting {
...
-    if (this._isInRickrollPeriod()) {
-      tokenPool = tokenPool.concat(RoomIdentifiers.rickroll)
-    }
-    if (this._isInValentinesPeriod()) {
-      tokenPool = tokenPool.concat(RoomIdentifiers.valentines)
-    }
-    if (this._isInLunarNewYearPeriod()) {
-      tokenPool = tokenPool.concat(RoomIdentifiers.lunarNewYear)
+    for (let period in PeriodicBadges) {
+      if (PeriodBadges[period].isHappening(new Date())) {
```
Comment (pianiststickman) [inline modules/token_collecting.js]: What do you think about moving the `new Date()` call out of the loop?

File: library/periodic_badges.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
+const PeriodicBadges = {
```
Comment (pianiststickman) [inline library/periodic_badges.js:1]: Think you may want to freeze and make this an export.

**Review by pianiststickman — CHANGES_REQUESTED** [review]
Thanks, nice refactoring!

Can you run `npm test` and fix the errors for me?

File: library/periodic_badges.js (hunk 2 above)
Comment (pianiststickman) [reply library/periodic_badges.js:1]: Oh, looks like you beat me to it already.

Comment (Two9A) [reply library/periodic_badges.js:1]: The wonder of Travis caught that one, yeah.

File: modules/token_collecting.js (hunk 1 above)
Comment (pianiststickman) [reply modules/token_collecting.js]: Actually I think we can just reuse `current`

Comment (Two9A) [reply modules/token_collecting.js]: Sensible. More sensible than redeclaring it under a different name...

**Review by pianiststickman — APPROVED** [review]
LGTM, I'll add the contributor badge when I patch this in.

**Pull request merged**